lib/vssh/libssh2.c

```diff
@@ -3132,7 +3133,9 @@ static CURLcode ssh_connect(struct connectdata *conn, bool *done)
     libssh2_session_callback_set(ssh->ssh_session,
                                  LIBSSH2_CALLBACK_SEND, sshsend.sendp);
   }
-  else if(conn->handler->protocol & CURLPROTO_SCP) {
+  else
```

Comment:
I don't understand the else statement here

Reply:
The `else` clause is part of the `if` statement that is now conditioned under `CURL_DISABLE_PROXY`. This is how the code looks after the change:
```c
#ifndef CURL_DISABLE_PROXY
  if(conn->http_proxy.proxytype == CURLPROXY_HTTPS) {
    ...
  }
  else
#endif /* CURL_DISABLE_PROXY */
  if(conn->handler->protocol & CURLPROTO_SCP) {
    conn->recv[FIRSTSOCKET] = scp_recv;
    conn->send[FIRSTSOCKET] = scp_send;
  }
  else {
    ...
```